protos/orchestrator_service.proto

```diff
@@ -671,6 +676,9 @@ service TaskHubSidecarService {
     // Rewinds an orchestration instance to last known good state and replays from there.
     rpc RewindInstance(RewindInstanceRequest) returns (RewindInstanceResponse);
 
+    // Restarts an orchestration instance.
+    rpc RestartInstance(RestartInstanceRequest) returns (CreateInstanceResponse);
```

Comment:
Although redundant, I recommend that we create a new `RestartInstanceResponse` message type for this operation. That way we can completely avoid breaking changes if we need to add any restart-specific fields to the response in the future.